contrib/config/vault/config.go

```diff
@@ -0,0 +1,91 @@
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-kratos/kratos/v2/config"
+	"github.com/hashicorp/vault/api"
+)
+
+// Option is etcd config option.
+type Option func(o *options)
+
+type options struct {
+	ctx  context.Context
+	path string
+}
+
+//  WithContext with registry context.
+func WithContext(ctx context.Context) Option {
+	return Option(func(o *options) {
+		o.ctx = ctx
+	})
+}
+
+// WithPath is config path
+func WithPath(p string) Option {
```

Comment:
WithNamespace

Reply:
The etcd config use the `withPath`， So I think it's better use the `withPath`.